docs/rfcs/bun-bundler.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * Bundler API
+ *
+ * This is a proposal for the JavaScript API for Bun's native bundler.
+ */
+
+import { BunPlugin, FileBlob } from "bun";
+
+export type JavaScriptLoader = "jsx" | "js" | "ts" | "tsx";
+export type MacroMap = Record<string, Record<string, string>>;
+export type Target = "bun" | "browser" | "node" | "neutral";
+export type Format = "iife" | "cjs" | "esm";
+export type JsxTransform = "transform" | "preserve" | "automatic";
+export type Loader =
+  | "base64"
+  | "binary"
+  | "copy"
+  | "css"
+  | "dataurl"
+  | "default"
+  | "empty"
+  | "file"
+  | "js"
+  | "json"
+  | "jsx"
+  | "text"
+  | "ts"
+  | "tsx";
+export type LogLevel = "verbose" | "debug" | "info" | "warning" | "error" | "silent";
+export type Charset = "ascii" | "utf8";
+
+export interface BundlerOptions {
+  // currently this only accepts a Loader string
+  // change: set an ext->loader map
+  loader?: { [k in string]: Loader };
+
+  // rename `platform` to `target`
+  target?:
```

Comment:
this isn't in scope right now, the v1 is not going to support down-transpiling

only the fields in platform